test/core/event_engine/posix/tcp_posix_socket_utils_test.cc

```diff
@@ -114,60 +112,59 @@ TEST(TcpPosixSocketUtilsTest, SocketMutatorTest) {
       sock = socket(AF_INET6, SOCK_STREAM, 0);
     }
     EXPECT_GT(sock, 0);
-    PosixSocketWrapper posix_sock(sock);
+    EventEnginePosixInterface fds;
+    FileDescriptor wrapped = fds.Adopt(sock);
     struct test_socket_mutator mutator;
     grpc_socket_mutator_init(&mutator.base, vtable);
 
     mutator.option_value = IPTOS_LOWDELAY;
     EXPECT_TRUE(
-        posix_sock
-            .SetSocketMutator(GRPC_FD_CLIENT_CONNECTION_USAGE,
-                              reinterpret_cast<grpc_socket_mutator*>(&mutator))
+        fds.SetSocketMutator(wrapped, GRPC_FD_CLIENT_CONNECTION_USAGE,
+                             reinterpret_cast<grpc_socket_mutator*>(&mutator))
             .ok());
     mutator.option_value = IPTOS_THROUGHPUT;
     EXPECT_TRUE(
-        posix_sock
-            .SetSocketMutator(GRPC_FD_CLIENT_CONNECTION_USAGE,
-                              reinterpret_cast<grpc_socket_mutator*>(&mutator))
+        fds.SetSocketMutator(wrapped, GRPC_FD_CLIENT_CONNECTION_USAGE,
+                             reinterpret_cast<grpc_socket_mutator*>(&mutator))
             .ok());
 
     mutator.option_value = IPTOS_RELIABILITY;
     EXPECT_TRUE(
-        posix_sock
-            .SetSocketMutator(GRPC_FD_CLIENT_CONNECTION_USAGE,
-                              reinterpret_cast<grpc_socket_mutator*>(&mutator))
+        fds.SetSocketMutator(wrapped, GRPC_FD_CLIENT_CONNECTION_USAGE,
+                             reinterpret_cast<grpc_socket_mutator*>(&mutator))
             .ok());
 
     mutator.option_value = -1;
     EXPECT_FALSE(
-        posix_sock
-            .SetSocketMutator(GRPC_FD_CLIENT_CONNECTION_USAGE,
-                              reinterpret_cast<grpc_socket_mutator*>(&mutator))
+        fds.SetSocketMutator(wrapped, GRPC_FD_CLIENT_CONNECTION_USAGE,
+                             reinterpret_cast<grpc_socket_mutator*>(&mutator))
             .ok());
     close(sock);
   };
   test_with_vtable(&mutator_vtable);
   test_with_vtable(&mutator_vtable2);
 }
 
-TEST(TcpPosixSocketUtilsTest, SocketOptionsTest) {
-  int sock = socket(PF_INET, SOCK_STREAM, 0);
-  if (sock < 0) {
-    // Try ipv6
-    sock = socket(AF_INET6, SOCK_STREAM, 0);
-  }
-  EXPECT_GT(sock, 0);
-  PosixSocketWrapper posix_sock(sock);
-  EXPECT_TRUE(posix_sock.SetSocketNonBlocking(1).ok());
-  EXPECT_TRUE(posix_sock.SetSocketNonBlocking(0).ok());
-  EXPECT_TRUE(posix_sock.SetSocketCloexec(1).ok());
-  EXPECT_TRUE(posix_sock.SetSocketCloexec(0).ok());
-  EXPECT_TRUE(posix_sock.SetSocketReuseAddr(1).ok());
-  EXPECT_TRUE(posix_sock.SetSocketReuseAddr(0).ok());
-  EXPECT_TRUE(posix_sock.SetSocketLowLatency(1).ok());
-  EXPECT_TRUE(posix_sock.SetSocketLowLatency(0).ok());
-  close(sock);
-}
+// Need to be discussed in a code review - do we need this at this granularity?
```

Comment:
Do we have test coverage for these APIs elsewhere?  If not, then it seems like we shouldn't get rid of this.

What's the down-side of having this?  Why would we *not* want it?